Student/Student/Submissions/SubmissionRubric/RubricViewController.swift

```diff
@@ -30,30 +30,37 @@ class RubricViewController: UIViewController {
     @IBOutlet weak var emptyViewLabel: UILabel!
     var models: [RubricViewModel] = []
     var presenter: RubricPresenter!
-    var cellHeightCache = [Int: CGFloat]()
+    var selectedRatingCache = [Int]()
+    var collectionViewDidSetupCells = false
 
     override func viewDidLoad() {
         super.viewDidLoad()
 
         emptyViewLabel.text = NSLocalizedString("There is no rubric for this assignment", comment: "")
 
-        setupCollectionView()
+        setupCollectionViewHeader()
         presenter.viewIsReady()
     }
 
-    func setupCollectionView() {
-        let id = String(describing: RubricCollectionViewCell.self)
-        let nib = UINib(nibName: id, bundle: Bundle(for: type(of: self)))
-        collectionView.register(nib, forCellWithReuseIdentifier: id)
-
+    func setupCollectionViewHeader() {
         let headerID = String(describing: GradeCircleReusableView.self)
         collectionView.register(GradeCircleReusableView.self, forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader, withReuseIdentifier: headerID)
     }
 
+    func setupCollectionViewCells() {
+        collectionViewDidSetupCells = true
+        let count = models.count
+        for i in 0..<count {
+            let id = String(describing: RubricCollectionViewCell.self)
+            let nib = UINib(nibName: id, bundle: Bundle(for: type(of: self)))
+            let cellID = "\(id)_\(i)"
```

Comment:
What's the purpose of having a unique re-use identifier? Is this to avoid re-using cells?

Reply:
yes

Reply:
Can we not reuse cells anymore this for this?

Reply:
avoiding re-use makes the animations easier.  When reuse is being used, you have to relayout the cell, so for example if you want to animate one of the circle buttons,  but you remove that view b/c it's being reused, the view being animated can be gone.  

How is my thinking on this? Can you think of a workaround this i may not have considered?

Reply:
That's fine. I'm just thinking that if we're not re-using cells then we might as well use stack views and avoid all the height calculation stuff we're doing for collection view.

At this point it doesn't make a lot of sense to rewrite to use stack views but I have to imagine that performance was the primary reason we went with stack views for rubrics in the first place.

Reply:
let me spend some time looking into this, i think you make some great points